projects/cardiovascular-patient-app/src/app/report-bp/custom-error-state-matcher.ts

```diff
@@ -0,0 +1,12 @@
+import { FormControl, FormGroupDirective, NgForm } from '@angular/forms';
+import { ErrorStateMatcher } from '@angular/material/core';
+
+export class CustomErrorStateMatcher implements ErrorStateMatcher {
+  isErrorState(control: FormControl | null, form: FormGroupDirective | NgForm | null): boolean {
+    const isSubmitted = form && form.submitted;
+    if (isSubmitted) {
+      return false;
+    }
+      return !!(control && control.invalid && (control.dirty || control.touched || isSubmitted));
+    }
```

Comment:
Please format the code. this indentation is confusing.
I don't think this logic is correct. It looks like it will never show a validation error after the form has been submitted once. If the user enters one BP and then starts to enter a second one, we still want validation to work correctly on the second BP entry.